### DISCUSSION TITLE
Installing code-server on a server with an old version of GLIBC

### DISCUSSION THREAD

**Discussion by Tobio89** [coder/code-server#6620] [Q&A]
Hi all,
I want to install code-server on my server instance, but I run into two major issues, being I don't have authorization to use sudo, and the server has an outdated version of GLIBC.

Is there a way to install code-server despite these limitations? An older version, maybe?

Thanks :)

**Comment by code-asher** [comment]
You can download and run it from anywhere.  For example, if you use our installation script:

```
curl -fsSL https://code-server.dev/install.sh | sh -s -- --method=standalone
``` 

Then it will install into `~/.local` (you can change this with `--prefix`).

Then you would run it with:

```
~/.local/bin/code-server
```

**Comment by code-asher** [comment]
Oh right forgot about the second part of your question, glibc.  An older version might work but you will need to check against your version of glibc.  Otherwise you will have to build code-server yourself.

**Reply by code-asher** [reply]
Or you can install it with the method above, then rebuild just the native Node modules with your own Node.  Something like this (untested):

```
cd ~/.local/lib/code-server-4.20.0/lib/vscode
npm rebuild
node ~/.local/lib/code-server-4.20.0
```

But you will still need the required system dependencies to do this (see the contributing document).
